state-transition/core/state_processor_genesis.go

```diff
@@ -61,9 +61,9 @@ func (sp *StateProcessor[
 
 	var eth1Data *ctypes.Eth1Data
 	eth1Data = eth1Data.New(
-		common.Root{},
-		0,
-		execPayloadHeader.GetBlockHash(),
+		ctypes.Deposits(deposits).HashTreeRoot(),
+		0,                      // Eth1 Deposit Count is not needed.
+		common.ExecutionHash{}, // Eth1 Block Hash is not needed.
```

Comment:
q: is the genesis execution payload header not part of the beacon state anymore with this? If so I believe we should restore this

Reply:
I thought about this but only wanted to make the minimal changes here. Since the deposit count is implicitly guaranteed and the execution block hash is also part of execution payload header already, CL already agrees on these data and we don’t need to explicitly set here